theano/gpuarray/dnn.py

```diff
@@ -386,7 +386,7 @@ def __init__(self, border_mode, subsample=(1, 1), conv_mode='conv',
 
     def make_node(self, kern_shape):
         if kern_shape.type.ndim != 1 or kern_shape.type.dtype != 'int64':
-            raise TypeError('kern must be 1D shape tensor')
+            raise TypeError('kern must be an int64 1D shape tensor')
```

Comment:
if kern_shape.type.ndim != 1 or kern_shape.dtype not in theano.tensor.basic.int_dtypes:
            raise TypeError('kern must be an int64 1D shape tensor')
        kern_shape = kern_shape.cast('int64')

Remove the change in the test.

We want to upcast int to int64 here to prevent user annoyance here.